src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/RazorCSharpInterceptionMiddleLayer.cs

```diff
@@ -3,30 +3,52 @@
 
 using System;
 using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.ExternalAccess.Razor;
-using Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Protocol.SemanticTokens;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.VisualStudio.Razor.LanguageClient;
 
 [Export(typeof(IRazorCSharpInterceptionMiddleLayer))]
 internal class RazorCSharpInterceptionMiddleLayer : IRazorCSharpInterceptionMiddleLayer
 {
-    private readonly InterceptionMiddleLayer _underlyingMiddleLayer;
+    private readonly LSPRequestInvoker _requestInvoker;
 
     [ImportingConstructor]
-    public RazorCSharpInterceptionMiddleLayer(InterceptorManager interceptorManager)
+    public RazorCSharpInterceptionMiddleLayer(LSPRequestInvoker requestInvoker)
     {
-        _underlyingMiddleLayer = new InterceptionMiddleLayer(interceptorManager, RazorLSPConstants.CSharpContentTypeName);
+        _requestInvoker = requestInvoker;
     }
 
     public bool CanHandle(string methodName)
-        => _underlyingMiddleLayer.CanHandle(methodName);
+        => methodName.Equals(Methods.WorkspaceSemanticTokensRefreshName);
 
     public Task HandleNotificationAsync(string methodName, JToken methodParam, Func<JToken, Task> sendNotification)
-        => _underlyingMiddleLayer.HandleNotificationAsync(methodName, methodParam, sendNotification);
+    {
+        // IMPORTANT: This API shape is old and the methodParam and sendNotification parameters are likely to be null
+        // as Roslyn moves away from Newtonsoft.Json. Do not use these parameters without designing a new API contract.
+        Debug.Assert(CanHandle(methodName), "Got a call to intercept a message we were not expecting");
+
+        var refreshParams = new SemanticTokensRefreshParams();
```

Comment:
```suggestion
        // Normally we use the LSPRequestInvoker when we're on the client side of things, and want
        // to send to a server, so at first glance this might seem redundant. However, we're actually
        // intercepting a message from the Roslyn server, so need to call into our server, so we can turn
        // around and call the client from our server, because its our server we want it to refresh
        // semantic tokens from.

        var refreshParams = new SemanticTokensRefreshParams();
```